src/appmixer/kafka/routes.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+const connections = require('./connections');
+
+module.exports = (context) => {
+
+    context.http.router.register({
+        method: 'POST',
+        path: '/connect/{mode}',
+        options: {
+            handler: async (req, h) => {
+
+                const { flowId, componentId } = req.payload;
+                const { mode } = req.params;
+                if (mode === 'consumer') {
+                    await connections.removeConnection({ flowId, componentId }, mode);
+                }
+                await connections.addConnection(context, req.payload, mode);
```

Comment:
Doesn't it create a new producer connection with every message sent?
I don't see this condition 
<img width="1207" alt="Plugin_approach__kafka_connector_by_harshaio_·_Pull_Request__77_·_clientIO_appmixer-connectors" src="https://github.com/clientIO/appmixer-connectors/assets/928772/bc287393-4a78-4c7a-8fab-c3b4f4195a93">

in the addProducerConnection function.